### DISCUSSION TITLE
Error with new update "AttributeError: module 'gradio' has no attribute 'Code'"

### DISCUSSION THREAD

**Discussion by OPGGodsend** [DominikDoom/a1111-sd-webui-tagcomplete#154] [General]
Error occurs during startup and the extension no longer works.

Error executing callback ui_settings_callback for C:\A1111 Web UI\stable-diffusion-webui\extensions\a1111-sd-webui-tagcomplete\scripts\tag_autocomplete_helper.py
Traceback (most recent call last):
  File "C:\A1111 Web UI\stable-diffusion-webui\modules\script_callbacks.py", line 143, in ui_settings_callback
    c.callback()
  File "C:\A1111 Web UI\stable-diffusion-webui\extensions\a1111-sd-webui-tagcomplete\scripts\tag_autocomplete_helper.py", line 297, in on_ui_settings
    }""", """Configure Hotkeys. For possible values, see https://www.w3.org/TR/uievents-key, or leave empty / set to 'None' to disable. Must be valid JSON.""", gr.Code, lambda: {"language": "json", "interactive": True}, section=TAC_SECTION))
AttributeError: module 'gradio' has no attribute 'Code'

**Comment by DominikDoom** [comment]
See #152, this is a version issue. Best way is to update your webui, but I just now added a fallback for the old gradio versions that don't have the gradio.Code module yet, so it should work without.
